File: chrome/test/chromeos/autotest/files/client/deps/chrome_test/chrome_test.py
#!/usr/bin/python

# Copyright (c) 2010 The Chromium OS Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

import common, commands, logging, os
from autotest_lib.client.bin import utils

version = 1

def setup(top_dir):
    if 'CHROME_ORIGIN' not in os.environ.keys() or \
       (os.environ['CHROME_ORIGIN'] != 'LOCAL_SOURCE' and
        os.environ['CHROME_ORIGIN'] != 'SERVER_SOURCE'):
        logging.info('Skipping Chrome test resource setup for non-local builds')
        return

    chrome_test_files = os.environ['SYSROOT'] + '/usr/local/autotest-chrome'
    logging.info('Configuring chrome test resources in %s' % top_dir)
    testsrc_dir = top_dir + '/test_src'

    # Copy test build outputs.
    utils.run('cp -ral %s %s' % (chrome_test_files, testsrc_dir))


pwd = os.getcwd()
utils.update_version(pwd + '/src', False, version, setup, pwd)

File: chrome/test/data/extensions/platform_apps/music_manager_private/device_id_value_returned/chrometest.js
// Copyright 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

var deviceIdMaybeUndefined;

function runTests() {
  chrome.test.runTests([
    function test() {
      chrome.musicManagerPrivate.getDeviceId(function(id) {
          console.log('Device ID=' + id);
          if (!id) {
            chrome.test.assertEq(true, deviceIdMaybeUndefined)
          } else {
            chrome.test.assertEq('string', typeof id);
            chrome.test.assertTrue(id.length >= 8);
          }
          chrome.test.succeed();
      });
    }
  ]);
}

window.onload = function() {
  chrome.test.getConfig(function(config) {
    console.log('customArg=' + config.customArg);
    deviceIdMaybeUndefined =
        (config.customArg === 'device_id_may_be_undefined');
    runTests();
  });
}
